narwhals/_pandas_like/utils.py

```diff
@@ -514,7 +513,7 @@ def reverse_translate_dtype(  # noqa: PLR0915
         if dtype_backend == "pandas-nullable":
             return "string"
         else:
-            return object
+            return str
```

Comment:
we should do `.astype(str)` here, and then just let pandas convert that to `object`. else, if we do `astype(object)`, then integers will stay integers